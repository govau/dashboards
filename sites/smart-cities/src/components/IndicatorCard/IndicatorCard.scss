@import '~style/main';

.wrapper {
  @include shadow;

  position: relative;
  background: $color-white;
  border-radius: 4px;
}
