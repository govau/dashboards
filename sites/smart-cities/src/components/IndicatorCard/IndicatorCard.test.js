import React from 'react';
import { shallow } from 'enzyme';
import IndicatorCard from './IndicatorCard';

jest.mock('./RangeCard/RangeCard', () => 'RangeCard');
jest.mock('./NumberCard/NumberCard', () => 'NumberCard');

const defaultProps = {
  colorName: 'jobs',
  indicatorId: 'population',
  color: '#fff',
};

it('should match Snapshot', () => {
  const component = shallow(<IndicatorCard {...defaultProps} />);
  expect(component.debug()).toMatchSnapshot();
});

it('should pass an indicator object with id to the child card', () => {
  const component = shallow(
    <IndicatorCard {...defaultProps} />
  );

  expect(component.prop('indicator')).toEqual(expect.objectContaining({
    id: 'population',
    name: 'Population',
  }));
});

it('should render a NumberCard when the city prop is present', () => {
  const component = shallow(
    <IndicatorCard
      {...defaultProps}
      city={{
        id: 'perth',
        indicators: {},
        name: 'Perth',
      }}
    />
  );

  expect(component.is('NumberCard')).toBe(true);
});

it('should render a RangeCard when the city prop is not present', () => {
  const component = shallow(
    <IndicatorCard
      {...defaultProps}
      city={null}
    />
  );

  expect(component.is('RangeCard')).toBe(true);
});
