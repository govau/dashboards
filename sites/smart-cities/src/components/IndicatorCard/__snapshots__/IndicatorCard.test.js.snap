// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `"<RangeCard colorName=\\"jobs\\" indicatorId=\\"population\\" color=\\"#fff\\" size=\\"small\\" className=\\"wrapper\\" indicator={{...}} />"`;
